### PR TITLE
RD-6364 support updating vault secret 

### PR DESCRIPTION
**Pull request by PatrycjaKubiczek**
## Description

RD-6364
- Add support for updating Vault Secret Provider

## Screenshots / Videos

Zrzut ekranu 2022-12-23 115832 

## Checklist

- [x] My code follows the [UI Code Style](https://cloudifysource.atlassian.net/l/c/x8fq902p).
- [x] I followed the [UI Testing Guidelines](https://cloudifysource.atlassian.net/l/cp/udCVfvrx) and verified that all tests/checks have passed.
- [x] I verified if that change requires any update in the [documentation](https://cloudifysource.atlassian.net/l/c/4XKtPocy).
- [x] I added proper labels to this PR.

## Tests

 

[Build Status](https://jenkins.cloudify.co/view/UI%20Health%20View/job/Stage-UI-System-Test/4198/)

### CODE REVIEW

File: widgets/secretProviders/src/UpdateSecretProviderButton.tsx [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+}
+
+const translateUpdateModal = Stage.Utils.composeT(translateSecretProviders, 'updateModal');
+
+const UpdateSecretProviderButton = ({ secretProvider, manager, onSubmit }: UpdateSecretProviderButtonProps) => {
+    const { errors, setErrors, clearErrors } = useErrors();
+
+    const [hostname, setHostname] = useInput(secretProvider?.connection_parameters?.host ?? '');
+    const [authorizationToken, setAuthorizationToken] = useInput(secretProvider?.connection_parameters?.token ?? '');
+    const [defaultPath, setDefaultPath] = useInput(secretProvider?.connection_parameters?.path ?? '');
```
Comment (Vorbert-Kruk) [inline widgets/secretProviders/src/UpdateSecretProviderButton.tsx:23]: WDYT about the following deduplication? 

```suggestion
   const connectionParameters = secretProvider?.connection_parameters;

    const [hostname, setHostname] = useInput(connectionParameters?.host ?? '');
    const [authorizationToken, setAuthorizationToken] = useInput(connectionParameters?.token ?? '');
    const [defaultPath, setDefaultPath] = useInput(connectionParameters?.path ?? '');
```

Comment (PatrycjaKubiczek) [reply widgets/secretProviders/src/UpdateSecretProviderButton.tsx]: good idea, implemented: https://github.com/cloudify-cosmo/cloudify-stage/pull/2351/commits/4e6cc0273c14b7a801a6a7313111c1eb589ae68c

File: widgets/secretProviders/src/UpdateSecretProviderButton.tsx [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+                        />
+                    </Form.Field>
+                </Form>
+            </Modal.Content>
+            <Modal.Actions>
+                <Button content={translateUpdateModal('buttons.cancel')} onClick={closeUpdateModal} />
+                <Button content={translateUpdateModal('buttons.update')} onClick={updateSecretProvider} primary />
+            </Modal.Actions>
+        </Modal>
+    );
```
[the remark this answers is not in the document]
Comment (PatrycjaKubiczek) [reply widgets/secretProviders/src/UpdateSecretProviderButton.tsx]: implemented: https://github.com/cloudify-cosmo/cloudify-stage/pull/2351/commits/4e6cc0273c14b7a801a6a7313111c1eb589ae68c :)

File: widgets/secretProviders/src/widget.utils.ts [hunk 3]
```diff
@@ -1 +1,27 @@
...
+
+export const validateModalForm = (form: Form, isCreateModal: boolean) => {
+    const newErrors: Record<string, string> = {};
+    if (isCreateModal) {
+        if (!form.providerName) {
+            newErrors.providerName = translateSecretProviders('form.errors.providerName.required');
+        } else if (!Consts.ID_REGEX.test(form.providerName)) {
+            newErrors.providerName = translateSecretProviders('form.errors.providerName.invalid');
+        }
+    }
```
[the remark this answers is not in the document]
Comment (PatrycjaKubiczek) [reply widgets/secretProviders/src/widget.utils.ts]: done: https://github.com/cloudify-cosmo/cloudify-stage/pull/2351/commits/4e6cc0273c14b7a801a6a7313111c1eb589ae68c :)

**Review by kubama — DISMISSED** [review]

File: test/cypress/integration/widgets/secret_providers_spec.ts [hunk 4]
```diff
@@ -48,4 +48,13 @@ describe('Secret Providers widget', () => {
...
         cy.contains('Secret_Provider_2').should('be.visible');
     });
+
+    it('should allow to update secret provider', () => {
+        cy.getWidget(widgetId).find('i[title="Update Secret Provider"]').click();
+        cy.typeToFieldInput('Vault Hostname', 'localhost_test');
+        cy.typeToFieldInput('Authorization Token', 'token_test');
+        cy.typeToFieldInput('Vault Default Path', 'path_test');
+        cy.clickButton('Update');
+        cy.getWidget(widgetId).find('td').eq(3).should('not.be.empty');
```
[the remark this answers is not in the document]
Comment (PatrycjaKubiczek) [reply test/cypress/integration/widgets/secret_providers_spec.ts]: https://github.com/cloudify-cosmo/cloudify-stage/pull/2351/commits/e338a9bdf28e8af64345b8c00c36c7ced457d516 done :)

[the remark this answers is not in the document]
Comment (PatrycjaKubiczek) [reply test/cypress/integration/widgets/secret_providers_spec.ts]: https://github.com/cloudify-cosmo/cloudify-stage/pull/2351/commits/e338a9bdf28e8af64345b8c00c36c7ced457d516 done :)

File: widgets/secretProviders/src/CreateSecretProviderModal.tsx [hunk 5]
```diff
@@ -27,42 +27,30 @@ const CreateSecretProviderModal = ({
...
     const handleSubmit = () => {
         clearErrors();
 
-        if (!validateForm()) {
+        if (!checkForm()) {
             return;
         }
 
         manager
             .doPut('/secrets-providers', {
```
[the remark this answers is not in the document]
Comment (PatrycjaKubiczek) [reply widgets/secretProviders/src/CreateSecretProviderModal.tsx:46]: after discussion on slack I update `doPatch` :) https://github.com/cloudify-cosmo/cloudify-stage/pull/2351/commits/51d9801a382ce14b06d42a4608672e3aad96dcc8

**Review by Vorbert-Kruk — APPROVED** [review]
LGTM 💪 
Thanks for addressing all comments 🎉
Awesome piece of work 🚀

**Pull request merged**